spec/functions/archive_go_md5_spec.rb

```diff
@@ -0,0 +1,16 @@
+require 'spec_helper'
+
+describe 'archive::go_md5' do
+  let(:example_md5) do
+    File.read(fixtures('checksum', 'gocd.md5'))
+  end
+
+  it { is_expected.not_to eq(nil) }
+
+  it 'retreives file md5' do
+    url = 'https://gocd.lan/path/file.md5'
+    uri = URI(url)
+    PuppetX::Bodeco::Util.stubs(:content).with(uri, username: 'user', password: 'pass').returns(example_md5)
+    is_expected.to run.with_params('user', 'pass', 'filea', url).and_return('283158c7da8c0ada74502794fa8745eb')
+  end
```

Comment:
Consider adding a test that demonstrates the error state and hits the `raise` in the function.

Reply:
I was aiming to limit my changes to just converting the function, but actually, I think you're right that the error handling code is broken.  Fixing that up should be simple enough.

Reply:
Fixes in https://github.com/voxpupuli/puppet-archive/pull/392/commits/9c5d6323d894acdf24d8a93bd1f63284fb966ad7